lib/changes/toggleList.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import { Document, Node } from 'slate';
+import Options from '../options';
+import { wrapInList } from '.';
+
+const LIST_TYPES = ['ol_list', 'ul_list'];
+
+type MappedNode = {
+    depth: number,
+    node: Node
+};
+
+function isList(blockType: string) {
+    return LIST_TYPES.includes(blockType);
+}
+
+function isItem(blockType: string) {
+    return blockType === 'list_item';
+}
+
+function filterListDescendants(node: Node) {
+    return isList(node.type) || isItem(node.type);
+}
+
+function mapListDescendants(document: Document) {
+    return (node: Node) => ({
+        depth: document.getDepth(node.key),
+        node
+    });
+}
+
+function sortListDescendants(a: MappedNode, b: MappedNode): -1 | 0 | 1 {
+    if (a.depth !== b.depth) {
+        return b.depth - a.depth;
+    }
+
+    if (a.node.type === b.node.type) {
+        return 0;
+    }
+
+    if (a.node.type === 'list_item') {
+        return -1;
+    }
+
+    return 1;
+}
+
+function unwrapMappedNodes(change: Change, mappedNode: MappedNode) {
+    return change.unwrapBlockByKey(mappedNode.node.key, undefined, {
+        normalize: false
+    });
```

Comment:
what would happen if normalize wasn't false? is there some custom normalization screwing thing up?